src/ngraph/runtime/gpu/gpu_cuda_kernel_emitters.hpp

```diff
@@ -88,25 +83,34 @@ namespace ngraph
                         writer, name + type_signature, op_name, data_types, sizeof...(inputs));
 
                     std::string kernel = writer.get_code();
-                    CudaFunctionPool::instance().set(name + type_signature, kernel);
+                    compiled_kernel = ctx->nvrtc_cache->set(name + type_signature, kernel);
                 }
 
                 //convert runtime ptr to driver api ptr
                 void* args_list[] = {&inputs..., &out, &count};
-                CUDA_SAFE_CALL(
-                    cuLaunchKernel(*CudaFunctionPool::instance().get(name + type_signature).get(),
-                                   count,
-                                   1,
-                                   1, // grid dim
-                                   1,
-                                   1,
-                                   1, // block dim
-                                   0,
-                                   NULL, // shared mem and stream
-                                   args_list,
-                                   0));             // arguments
+                CUDA_SAFE_CALL(cuLaunchKernel(*compiled_kernel.get(),
+                                              count,
+                                              1,
+                                              1, // grid dim
+                                              1,
+                                              1,
+                                              1, // block dim
+                                              0,
+                                              NULL, // shared mem and stream
+                                              args_list,
+                                              0));  // arguments
                 CUDA_SAFE_CALL(cuCtxSynchronize()); // Retrieve and print output.
             }
+
+            void emit_reshape(const std::string& name,
```

Comment:
since we are changing the order anyway, does it make sense to put it in alphabet order?